code/obj/machinery/singularity.dm

```diff
@@ -21,16 +21,18 @@ Contains:
 	density = 1
 	mats = 250
 	var/bhole = 0 // it is time. we can trust people to use the singularity For Good - cirr
-/* no
-/obj/machinery/the_singularitygen/New()
-	..()
-*/
+
 /obj/machinery/the_singularitygen/process()
-	var/checkpointC = 0
-	for (var/obj/X in orange(4,src))
-		if (istype(X, /obj/machinery/containment_field))
-			checkpointC ++
-	if (checkpointC >= 20)
+	var/goodgenerators = 0 //ensures that there are 4 generators in place with at least 2 links. note that false positives are very possible and will result in a loose singularity
+	var/smallestdimension = 13//determines the radius of the produced singularity,starts higher than is possible
+
+	for_by_tcl(gen, /obj/machinery/field_generator)//this loop checks for valid field generators
+		if(get_dist(gen,loc)<10)
```

Comment:
10 is a bit of a magic number here. Shouldn't this be based off some macro constant that determines the max singularity size?